From 5410fe404a8b7c3af2d33b18bd8436fffac12717 Mon Sep 17 00:00:00 2001
From: Timo Rothenpieler <timo@rothenpieler.org>
Date: Tue, 7 Jan 2025 18:18:02 +0100
Subject: [PATCH 11/21] UPSTREAM: avformat/flvdec: fix potential premature
 return on audio MultichannelConfig

(cherry picked from commit 0ed34467382a35dd27da7124fae99e84eee469a5)
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 libavformat/flvdec.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libavformat/flvdec.c b/libavformat/flvdec.c
index e8b5ea954365..043c977e73de 100644
--- a/libavformat/flvdec.c
+++ b/libavformat/flvdec.c
@@ -1273,7 +1273,7 @@ static int flv_update_video_color_info(AVFormatContext *s, AVStream *st)
 static int flv_read_packet(AVFormatContext *s, AVPacket *pkt)
 {
     FLVContext *flv = s->priv_data;
-    int ret, i, size, flags;
+    int ret = 0, i, size, flags;
     int res = 0;
     enum FlvTagType type;
     int stream_type=-1;
@@ -1611,7 +1611,7 @@ retry_duration:
 
                 av_log(s, AV_LOG_DEBUG, "Set channel data from MultiChannel info.\n");
 
-                goto leave;
+                goto next_track;
             }
         } else if (stream_type == FLV_STREAM_TYPE_VIDEO) {
             int sret = flv_set_video_codec(s, st, codec_id, 1);
@@ -1757,6 +1757,7 @@ retry_duration:
             return ret;
         res = FFERROR_REDO;
 
+next_track:
         if (track_size) {
             av_log(s, AV_LOG_WARNING, "Track size mismatch: %d!\n", track_size);
             avio_skip(s->pb, track_size);
@@ -1766,7 +1767,6 @@ retry_duration:
         if (!size)
             break;
 
-next_track:
         if (multitrack_type == MultitrackTypeOneTrack) {
             av_log(s, AV_LOG_ERROR, "Attempted to read next track in single-track mode.\n");
             ret = FFERROR_REDO;
-- 
2.48.1.windows.1

